schemas/pci/pci-bus.yaml

```diff
@@ -22,139 +23,202 @@ description: |
+        minItems: 2
+        maxItems: 2
+        items:
+          maximum: 255
+
+      "#interrupt-cells":
+        const: 1
+
+      interrupt-map: true
+#        minItems: 1
+#        maxItems: 88    # 22 IDSEL x 4 IRQs
+#        items:
+#          minItems: 6   # 3 addr cells, 1 PCI IRQ cell, 1 phandle, 1+ parent addr and IRQ cells
+#          maxItems: 16
+
+      interrupt-map-mask:
+        items:
+          - description: PCI high address cell
+            minimum: 0
+            maximum: 0xf800
+          - description: PCI mid address cell
+            const: 0
+          - description: PCI low address cell
+            const: 0
+          - description: PCI IRQ cell
+            minimum: 0
+            maximum: 7
+
+    patternProperties:
+      ".*":
+        allOf:
+          - if:
+              type: object
+              properties:
+                device_type:
+                  const: pci
+            then:
+              allOf:
+                - $ref: "#/$defs/pci-function"
+              properties:
+                $nodename:
+                  pattern: "^[a-zA-Z][a-zA-Z0-9,+\\-._]{0,63}@1?[0-9a-f](,[0-7])?$"
+          - if:
+              type: object
+              properties:
+                compatible:
+                  contains:
+                    const: "pciclass,0604"
+            then:
+              $ref: "#/$defs/pci-bridge"
+
+      "@1?[0-9a-f](,[0-7])?$":
+        $ref: "#/$defs/pci-function"
+
+    required:
+      - device_type
+
+
+  pci-master-bus:
+    title: PCI master bus node
+
+    description: |
+      Node which represents PCI host bridge, root of the PCI bus segment group.
+      Subnodes represents top level PCI functions.
+
+    allOf:
+      - $ref: "#/$defs/pci-child-bus"
+
+    properties:
+      ranges:
+        oneOf:
+          - $ref: /schemas/types.yaml#/definitions/flag
+          - minItems: 1
+            maxItems: 32    # Should be enough
+            items:
+              minItems: 5
+              maxItems: 7
+              additionalItems: true
+              items:
+                - enum:
+                    - 0x01000000
+                    - 0x02000000
+                    - 0x03000000
+                    - 0x42000000
+                    - 0x43000000
+                    - 0x81000000
+                    - 0x82000000
+                    - 0x83000000
+                    - 0xc2000000
+                    - 0xc3000000
+
+      dma-ranges:
+        oneOf:
+          - type: boolean
+          - minItems: 1
+            maxItems: 32    # Should be enough
+            items:
+              minItems: 5
+              maxItems: 7
+              additionalItems: true
+              items:
+                - enum:
+                    - 0x02000000
+                    - 0x03000000
+                    - 0x42000000
+                    - 0x43000000
+
+      "#address-cells":
+        const: 3
+
+      "#size-cells":
+        const: 2
```

Comment:
Every bridge node needs these and ranges. Not sure on dma-ranges. Technically possible I'd guess, but unlikely.

Reply:
`dma-ranges` is not marked as required.